daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/SequenceParserBases.scala

```diff
@@ -186,6 +186,9 @@ abstract class SequenceParserBase(
               if (ais ne Done) {
                 pstate.mpstate.moveOverOneArrayIndexOnly()
               }
+              if (resultOfTry == ParseAttemptStatus.NormalRep) {
```

Comment:
This is one of very few cases when I'd actually prefer the code to use the negative and say:
```
resultOfTry != ParseAttemptStatus.AbsentRep
```
Then you can add a comment explaining that if the rep is AbsentRep, then no element is added to the array, but in every other case we do add an element to the array, and so we count it.

Reply:
These changes have been replaced with counting the children in the array node.